liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -225,10 +225,15 @@ lhDynFlags _ hscEnv =
 -- TODO: We shouldn't rely on exports to find out what's in a ModGuts
 -- https://github.com/ucsd-progsys/liquidhaskell/pull/2388#issuecomment-2411418479
 desugarerDynFlags :: DynFlags -> DynFlags
-desugarerDynFlags df = df
+desugarerDynFlags df = (foldl gopt_unset df disabledOpts)
     { debugLevel   = 1                  -- To keep source note ticks
     , backend      = interpreterBackend -- To export everything
     }
+  where
+    disabledOpts =
+      [ Opt_EnableRewriteRules
```

Comment:
Given that the failure is not too easy to reproduce, more of an explanation would be needed to communicate why this can't be removed.